CONTRIBUTING.md

```diff
@@ -69,6 +69,224 @@ You can control the execution of long-running tests over the `INCLUDE_SLOW_TESTS
 To change this setup, open (or create) the `.env` file in the root folder.
 Edit or add the environment variable there as follows: `INCLUDE_SLOW_TESTS=true`.
 
+### Publishing
+
+> [!NOTE] Projects are marked as publishabel by adding a target named `publishabel`.
+> Those libraries will have dynamic targets to publish and install the package.
+
+Every publishable project in the monorepo has the following targets:
+
+- [`publish`](./tools/src/publish/README.md#publish) - publish the package to the local registry
+- [`npm-check`](./tools/src/npm/README.md#npm-check) - check if the package is installed in registry
+- [`npm-install`](./tools/src/npm/README.md#npm-install) - install package.
+- [`npm-uninstall`](./tools/src/npm/README.md#npm-uninstall) - uninstall package form project
+
+The following steps are necessary to publish a package:
+
+1. `nx run <project-name>:npm-check` - check if the package is not already published
+2. `nx run <project-name>:publish --nextVersion=<version>` - publish package (login required)
+3. `nx run <project-name>:npm-check` - check if the package is published
+
+### E2e testing
+
+> [!NOTE] Projects that need verdaccio are identified over the `e2e` target.
+> Those libraries will have dynamic targets to start verdaccio and test the package.
+
+All e2e tests use verdaccio to test the build artefact in a real registry.
+
+Every e2e project in the monorepo has the following targets:
+
+- [`start-verdaccio`](./tools/src/verdaccio/README.md#start-verdaccio) - start a local registry
+
+**Running e2e tests for a given project:**
+
+Every project in the monorepo that has e2e tests follows the project naming pattern: `<project-name>-e2e`.
+
+Examples:
+
+- `npx nx e2e cli-e2e` - run e2e tests for the cli project
+- `npx nx e2e cli-e2e --skipNxCache` - pass Nx CLI arguments
+- `npx nx run-many -t e2e` - run all e2e tests
+
+#### E2e testing process
```

Comment:
This chapter describes the essence of the PR and is perfect to get a big picture